specification/redis/Cache.Management/RedisResource.tsp

```diff
@@ -0,0 +1,228 @@
+  delete is RedisResourcesOps.DeleteWithoutOkAsync<
+    RedisResource,
+    Response = ArmDeletedResponse | ArmDeleteAcceptedLroResponse | ArmDeletedNoContentResponse
+  >;
+
+  /**
+   * Lists all Redis caches in a resource group.
+   */
+  listByResourceGroup is ArmResourceListByParent<
+    RedisResource,
+    Response = ArmResponse<RedisListResult>
+  >;
+
+  /**
+   * Gets all Redis caches in the specified subscription.
+   */
+  listBySubscription is ArmListBySubscription<
+    RedisResource,
+    Response = ArmResponse<RedisListResult>
+  >;
+
+  /**
+   * Gets any upgrade notifications for a Redis cache.
+   */
+  @get
+  listUpgradeNotifications is RedisResourcesOps.ActionSync<
+    RedisResource,
+    void,
+    ArmResponse<NotificationListResponse>,
+    Parameters = {
+      /**
+       * how many minutes in past to look for upgrade notifications
+       */
+      @query("history")
+      history: float64;
+    }
+  >;
+
+  /**
+   * Retrieve a Redis cache's access keys. This operation requires write permission to the cache resource.
+   */
+  listKeys is RedisResourcesOps.ActionSync<
+    RedisResource,
+    void,
+    ArmResponse<RedisAccessKeys>
+  >;
+
+  /**
+   * Regenerate Redis cache's access keys. This operation requires write permission to the cache resource.
+   */
+  regenerateKey is RedisResourcesOps.ActionSync<
+    RedisResource,
+    RedisRegenerateKeyParameters,
+    ArmResponse<RedisAccessKeys>
+  >;
+
+  /**
+   * Reboot specified Redis node(s). This operation requires write permission to the cache resource. There can be potential data loss.
+   */
+  forceReboot is RedisResourcesOps.ActionSync<
+    RedisResource,
+    RedisRebootParameters,
+    ArmResponse<RedisForceRebootResponse>
+  >;
+
+  /**
+   * Import data into Redis cache.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-post-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @action("import")
+  importData is RedisResourcesOps.ActionAsync<
+    RedisResource,
+    ImportRDBParameters,
+    void,
+    Response = OkResponse | ArmAcceptedLroResponse | ArmNoContentResponse
+  >;
+
+  /**
+   * Export data from the redis cache to blobs in a container.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-post-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @action("export")
+  exportData is RedisResourcesOps.ActionAsync<
+    RedisResource,
+    ExportRDBParameters,
+    void,
+    Response = OkResponse | ArmAcceptedLroResponse | ArmNoContentResponse
+  >;
+
+  /**
+   * Deletes all of the keys in a cache.
+   */
+  @action("flush")
+  flushCache is ArmResourceActionAsync<
+    RedisResource,
+    void,
+    Azure.ResourceManager.CommonTypes.OperationStatusResult
```

Comment:
What is actually returned by the service?

Reply:
@TimLovellSmith Do you know if the OperationStatusResult is the real response?